far/config.cpp

```diff
@@ -1230,7 +1230,7 @@ void Options::SetFilePanelModes()
 
 		if (DeleteMode)
 		{
-			const auto SwitchToAnotherMode = [&](panel_ptr p)
+			const auto SwitchToAnotherMode = [&](const panel_ptr& p)
```

Comment:
We pass shared pointers by value for a reason.
Window handling in Far is somewhat complicated and certain operations can remove individual windows from their container, thus destroying the corresponding pointers if they're passed by a const ref.
On the contrary, when we pass by value, it is guaranteed that the pointer will stay alive until the end of the function.

A simplified example:
https://godbolt.org/z/ej3rzda3P

Unless you're absolutely sure that the function and all the other functions it may call won't invalidate the passed pointer and *it will be the case forever*, it's better to leave it as is.